### ISSUE TITLE
During matrix.org upgrade, msg sent from vector failed with 50x and vanished

### ISSUE THREAD

**Issue by matrixbot** [vector-im/riot-web#628]
Created by @ matthew:matrix.org.

First it went red, then there was a 'Click here to resend unsent messages' prompt; next message went to grey and was eventually resent - but the 'Click here to resend' prompt got stuck on (until sending the next message) and the original message disappeared entirely from the timeline and was never sent. It was stuck in state not_sent in the room's _txnToEvent dict however

**Comment by Kegsay** [comment]
So what I've been able to diagnose from this is:
 - The event was added to the timeline before it was sent (it exists in `Room._txnToEvent` dict)
 - At some point, the event was either *removed* or *clobbered*.

There are very few places where we remove elements in the timeline. These places are:
 - `/sync` returning a `limited: true` response for a room (this nukes the entire timeline)
 - `Room.removeEvents` is called. This is only called in 1 file: `MatrixClient._sendEvent` and only when the `promise` for the HTTP resolve has *resolved*. I dug into the scheduler/HTTP API and didn't notice anything which would cause it to resolve incorrectly.

Therefore, I believe that the event is being clobbered. Again, there are very few places where we aren't just pushing and unshifting (to front of array) on the timeline. These places are:
 - When `Room.addEvents` is called with a `duplicateStrategy` (we don't do this anywhere in the code anymore!)

So I'm missing something (searched JS and React SDK) here. For future digging, the main facts are:
 - There is an event in `Room._txnToEvent` which is *not* in the `Room.timeline`. This should be impossible.
